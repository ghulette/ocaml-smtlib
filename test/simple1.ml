open Easy_smt.Smtlib
open Smtlib_test

let test1 solv =
  declare_const solv (Id "x") int_sort;
  declare_const solv (Id "y") int_sort;
  let x = const "x" in
  let y = const "y" in
  assert_ solv (equals x (int_to_term 5));
  assert_ solv (equals y (add x (int_to_term 1)))

let () =
  run_test test1
