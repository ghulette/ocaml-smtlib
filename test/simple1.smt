(declare-const x Int)
(declare-const y Int)
(assert (= x 5))
(assert (= y (+ x 1)))
(check-sat)
(get-model)
