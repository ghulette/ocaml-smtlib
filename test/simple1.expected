sat
y : Int := 6
x : Int := 5
